### PR TITLE
Conversion of a part of page to react

### PR DESCRIPTION
**Pull request by kamontia**

### CODE REVIEW

File: app/javascript/packs/app.tsx [hunk 1]
```diff
@@ -1,17 +1,45 @@
 import React, { Component } from 'react'
 
+interface TaskList {
+  task: Task[],
+  isLoading: Boolean
+}
 
+interface Task {
```
Comment (chaspy) [inline app/javascript/packs/app.tsx:8]: Use this

```ruby
create_table "todos", force: :cascade do |t|
  t.string "title"
  t.string "description"
  t.integer "user_id"
  t.boolean "finished"
  t.datetime "created_at", precision: 6, null: false
  t.datetime "updated_at", precision: 6, null: false
end
```

Comment (kamontia) [reply app/javascript/packs/app.tsx]: @chaspy app/javascript/type.tsx

File: app/javascript/type.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    created_at: Date
+    updated_at: Date
+}
+export type Request = {
+    user_id: Number
+    title: string
+    description: string
+    finished: Boolean
+    created_at: Date
+    updated_at: Date
```
Comment (chaspy) [inline app/javascript/type.tsx:15]:
```suggestion
```

File: app/views/todos/index.html.erb [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
+
+<% if user_signed_in? %>
 <section>
   <%= javascript_pack_tag 'hello_react' %>
 </section>
-
+<% else %>
+  <%= link_to 'Sign up with GitHub', user_github_omniauth_authorize_path %>
+<% end %>
+<!--
```
Comment (chaspy) [inline app/views/todos/index.html.erb:11]: Do not remove whole features for now.
Let's migrate to typescript gradually.

Comment (kamontia) [reply app/views/todos/index.html.erb]: Update to remain rails views.

**Comment by kamontia** [comment]
@chaspy  Please review and merge

**Pull request merged**